tests/unit/parsers/test.manifestjson.js

```diff
@@ -102,6 +102,125 @@ describe('ManifestJSONParser', () => {
     expect(warnings[0].code).toEqual('IGNORED_APPLICATIONS_PROPERTY');
   });
 
+  it.each([
+    {
+      title: 'hidden and browser_action',
+      manifest: {
+        hidden: true,
+        browser_action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden and action',
+      manifest: {
+        hidden: true,
+        action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden and page_action',
+      manifest: {
+        hidden: true,
+        page_action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden, browser_action and page_action',
```

Comment:
Nit, technically in mv3 there will be a similar but slightly different combination, 'hidden, action and page_action' at least at the moment, given that adding one more case is pretty cheap thanks to how you wrote this test cases, I'd say let's just add the additional case for now.